beacon-chain/core/helpers/rewards_penalties.go

```diff
@@ -284,122 +263,101 @@ func TotalRewardWithReserveUsage(s state.ReadOnlyBeaconState) (uint64, uint64) {
-//		target_deposit,
-//		pending_deposit,
-//		exit_deposit,
-//		target_change_rate,
-//		threshold
-//	):
-//		deposit_delta = pending_deposit - exit_deposit
-//
-//	 	future_deposit = deposit + deposit_delta
-//	 	error_rate = abs(future_deposit - target_deposit) / target_deposit
+// def process_reward_adjustment_factor(state: BeaconState) -> None:
+// _, future_total_active_balance = get_balance_with_queue(state)
+// target_deposit = get_target_deposit(state)
 //
-//	 	scale = max(min_change_rate, min(1.0, error_rate / threshold))
-//
-//	 	if future_deposit > target_deposit:
-//	     	return -target_change_rate * scale
-//	 	else:
-//	     	return target_change_rate * scale
-//
-//	def process_feedback(epoch, deposit, pending_deposit, exit_deposit):
-//
-//		target_deposit = TARGET_DEPOSIT_PLAN[epoch]
-//
-//		bias_delta = calc_feedback(
-//			deposit,
-//			target_deposit,
-//			pending_deposit,
-//			exit_deposit,
-//			threshold=FEEDBACK_THRESHOLD
-//			target_change_rate = TARGET_CHANGE_RATE
-//		)
-//
-//		return bias_delta
-//
-//	def process_validator_reward(epoch, parent, deposit, pending_deposit, exit_deposit):
-//
-//		prev_bias = parent.bias
-//
-//		bias_delta = process_feedback(epoch, deposit, pending_deposit, exit_deposit)
-//		bias = prev_bias + bias_delta
-//		bias = max(0, bias)
-//		bias = min(bias, MAX_BOOST_YIELD[epoch])
-//
-//		return bias
-func CalculateRewardAdjustmentFactor(state state.ReadOnlyBeaconState) (uint64, error) {
-	cfg := params.BeaconConfig()
-	epoch := slots.ToEpoch(state.Slot())
+// if future_total_active_balance > target_deposit:
+// decrease_reward_adjustment_factor(state, REWARD_ADJUSTMENT_FACTOR_DELTA)
+// elif future_total_active_balance < target_deposit:
+// increase_reward_adjustment_factor(state, REWARD_ADJUSTMENT_FACTOR_DELTA)
+func ProcessRewardAdjustmentFactor(state state.BeaconState) error {
 	futureDeposit, err := TotalBalanceWithQueue(state)
 	if err != nil {
-		return 0, err
+		return err
 	}
 	targetDeposit := TargetDepositPlan(time.NextEpoch(state))
 
-	// Using big integers for precise calculation
-	bigFutureDeposit := big.NewInt(int64(futureDeposit)) // lint:ignore uintcast -- changeRate will not exceed int64 because of total issuance.
-	bigTargetDeposit := big.NewInt(int64(targetDeposit)) // lint:ignore uintcast -- changeRate will not exceed int64 because of total issuance.
-	bigRewardPrecision := big.NewInt(int64(cfg.RewardFeedbackPrecision))
-
-	// Calculate the gap and error rate to make mitigating factor
-	gap := new(big.Int).Abs(new(big.Int).Sub(bigFutureDeposit, bigTargetDeposit))
-	numerator := new(big.Int).Mul(gap, bigRewardPrecision)
-	errRate := new(big.Int).Div(numerator, bigTargetDeposit)
-	mitigatingFactor := big.NewInt(int64(mathutil.Max(1000000, mathutil.Min(cfg.RewardFeedbackPrecision, errRate.Uint64()*cfg.RewardFeedbackThresholdReciprocal)))) // lint:ignore uintcast -- errRate will not exceed int64 because of truncation.
+	if futureDeposit >= targetDeposit {
```

Comment:
It seems like reward adjustment factor doesn't change where `futureDeposit == targetDeposit`.

Reply:
```python
def process_reward_adjustment_factor(state: BeaconState) -> None:
    _, future_total_active_balance = get_balance_with_queue(state)
    target_deposit = get_target_deposit(state)

    if future_total_active_balance > target_deposit:
        decrease_reward_adjustment_factor(state, REWARD_ADJUSTMENT_FACTOR_DELTA)
    elif future_total_active_balance < target_deposit:
        increase_reward_adjustment_factor(state, REWARD_ADJUSTMENT_FACTOR_DELTA)
```